### PR TITLE
Feature/8  view and edit activity

### LINKED ISSUE #8: Implement View and Edit Activity screen
**Describe the solution you'd like**
Pressing the "Add Activity" / "Edit Activity" buttons at the Trip Itinerary screen (in an activity group) redirects to a new screen to view or edit an activity.

Currently, documents are not implemented for activities.

**Additional context**
None.

IMG_0352
IMG_0351

### PR DESCRIPTION
**Pull request by whiteclouds7**
closes #8

I've decided to leave the save button at the bottom of the page, instead of moving it to the header. I think it's more consistent this way.

Unfortunately I had to user an older version of the datepicker library, as the current didn't work with expo version.

grafik
grafik
grafik

### CODE REVIEW

File: src/screens/Itinerary/TripItinerary.tsx [hunk 1]
```diff
@@ -45,6 +45,14 @@ export default function TripItinerary(props: Props): JSX.Element {
         console.log("Add activity group button pressed");
     };
 
+    const onAddActivity = (tripId: string, activityGroupId: string): void => {
+        props.navigation.navigate(Routes.VIEW_ADD_EDIT_ACTIVITY, {
+            tripId: tripId,
+            activityGroupId: activityGroupId,
+            mode: "add",
```
Comment (valentingc) [inline src/screens/Itinerary/TripItinerary.tsx]: you could use the "mode" enum from ViewEditActivity here

File: src/components/ActivityCard/ActivityCard.tsx [hunk 2]
```diff
@@ -149,9 +152,18 @@ export default function ActivityCard(props: ActivityCardProps): JSX.Element {
         }`;
     };
 
+    const openActivityDetails = (): void => {
+        navigation.navigate(Routes.VIEW_ADD_EDIT_ACTIVITY, {
+            activityId: props.id,
+            tripId: props.tripId,
+            activityName: props.name,
+            mode: "view",
```
Comment (valentingc) [inline src/components/ActivityCard/ActivityCard.tsx]: you could use the enum here

**Review by valentingc — CHANGES_REQUESTED** [review]
Great work! Works fine for me. Maybe we need to align the style of activity groups to your style too - I like the edit via header button, seems more fluent. I added two comments that should be a quick fix.

File: src/screens/Itinerary/TripItinerary.tsx (hunk 1 above)
Comment (whiteclouds7) [reply src/screens/Itinerary/TripItinerary.tsx]: mode was originally a type, but i've changed it to an enum now :), more developer friendly

**Comment by whiteclouds7** [comment]
@alexw-at can you check if everything works for android? thx

**Comment by whiteclouds7** [comment]
@valentingc i guess its probably cleaner if we add the edit button in a seperate PR, what do you think?

File: src/screens/Itinerary/TripItinerary.tsx (hunk 1 above)
Comment (valentingc) [reply src/screens/Itinerary/TripItinerary.tsx]: Oh sorry, I thought it was an enum.. My bad! 😥

**Comment by valentingc** [comment]
> @valentingc i guess its probably cleaner if we add the edit button in a seperate PR, what do you think?

Agree!

**Review by valentingc — APPROVED** [review]

**Pull request merged**